### PR TITLE
[plugin.video.slingtv] 1.0.0

### PR DESCRIPTION
**Pull request by Lunatixz**
### Checklist:

- [x] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [x] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

**Comment by Lunatixz** [comment]
@enen92 can you review the Travis errors? Should be nothing wrong with using 'filter' and 'izip'?

**Comment by Rechi** [comment]
> Should be nothing wrong with using 'filter' and 'izip'?

That's just an info, but `ERROR: Schema validation failed for the following points: xbmc.addon.metadata`.

**Comment by Lunatixz** [comment]
@Rechi the url is correct; however its a private development repository... maybe URL's issues should be marked as a warning? not an error?

**Comment by Rechi** [comment]
Please read what is actually reported as `ERROR`, `WARN` and `INFO` don't fail the check.

**Comment by Lunatixz** [comment]
@Rechi np, I'm reading over my mobile and formatting its problematic.

File: plugin.video.slingtv/addon.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        <import addon="script.module.xmltodict" version="0.11.0"/>
+    </requires>
+    <extension point="xbmc.python.pluginsource" library="default.py">
+        <provides>video</provides>
+    </extension>
+    <extension point="xbmc.addon.metadata">
+        <description lang="en_GB"></description>
+        <platform>all</platform>
+        <license>GNU GENERAL PUBLIC LICENSE. Version 3, June 2007</license>
+        <source>https://github.com/Lunatixz/plugin.video.slingtv</source>
```
Comment (Rechi) [inline plugin.video.slingtv/addon.xml]: This repo doesn't exist.

File: plugin.video.slingtv/resources/lib/slingtv.py [hunk 2]
```diff
@@ -0,0 +1,791 @@
...
+            fanart = channel_logo
+
+        infoArt = {
+            'thumb': thumb,
+            'poster': thumb,
+            'fanart': fanart,
+            'logo': channel_logo,
+            'clearlogo': channel_logo
+        }
+        print 'setAssetInfo', infoLabels, infoArt
```
Comment (Rechi) [inline plugin.video.slingtv/resources/lib/slingtv.py]:
```suggestion
        xbmc.log('setAssetInfo: {}, {}'.format(infoLabels, infoArt))
```

File: plugin.video.slingtv/resources/lib/slingtv.py [hunk 3]
```diff
@@ -0,0 +1,791 @@
...
+
+        USER_DMA = region['USER_DMA']
+        USER_OFFSET = region['USER_OFFSET']
+
+        channelURL = '%s/cms/publish3/domain/channels/v4/%s/%s/%s/1.json' % \
+                     (self.endPoints['environments']['production']['cms_url'], USER_OFFSET, USER_DMA,
+                      base64.b64encode(LEGACY_SUBS.replace('+', ',')))
+        subpacks = self.getURL(channelURL)['subscriptionpacks']
+        for subpack in subpacks:
+            print subpack
```
Comment (Rechi) [inline plugin.video.slingtv/resources/lib/slingtv.py]:
```suggestion
            xbmc.log(subpack)
```

File: plugin.video.slingtv/resources/lib/login.py [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Sling.TV.  If not, see <http://www.gnu.org/licenses/>.
+
+# -*- coding: utf-8 -*-
+import os, sys, time, _strptime, datetime, re, traceback, pytz, calendar, random
+import urlparse, urllib, urllib2, socket, json, requests, base64, inputstreamhelper
+import xbmc, xbmcgui, xbmcplugin, xbmcaddon, xbmcvfs
+from itertools import izip, cycle
```
Comment (Rechi) [inline plugin.video.slingtv/resources/lib/login.py]: izip is not available in python 3

File: plugin.video.slingtv/resources/lib/login.py [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+        REAL_SETTINGS.setSetting('subscriber_id', '')
+        REAL_SETTINGS.setSetting('user_dma', '')
+        REAL_SETTINGS.setSetting('user_subs', '')
+        REAL_SETTINGS.setSetting('legacy_subs', '')
+
+
+    def xor(self, data, key='INSECURE', encode=False):
+        if not encode:
+            data = base64.decodestring(data)
+        xored = ''.join(chr(ord(x) ^ ord(y)) for (x, y) in izip(data, cycle(key)))
```
Comment (Rechi) [inline plugin.video.slingtv/resources/lib/login.py]:
```suggestion
        xored = ''.join(chr(ord(x) ^ ord(y)) for (x, y) in zip(data, cycle(key)))
```

File: plugin.video.slingtv/resources/lib/slingtv.py [hunk 6]
```diff
@@ -0,0 +1,791 @@
...
+# along with Sling.TV.  If not, see <http://www.gnu.org/licenses/>.
+
+# -*- coding: utf-8 -*-
+import os, sys, time, _strptime, datetime, re, traceback, pytz, calendar, random, hashlib, xmltodict
+import urlparse, urllib, urllib2, socket, json, requests, base64, inputstreamhelper
+import xbmc, xbmcgui, xbmcplugin, xbmcaddon, xbmcvfs
+                            
+from simplecache import SimpleCache, use_cache
+from resources.lib.globals import *
+from login import UserClient
```
Comment (Rechi) [inline plugin.video.slingtv/resources/lib/slingtv.py]:
```suggestion
from .login import UserClient
```

File: plugin.video.slingtv/addon.xml (hunk 1 above)
Comment (Lunatixz) [reply plugin.video.slingtv/addon.xml]: Odd, it's a valid URL; however, it's marked private.

File: plugin.video.slingtv/resources/lib/slingtv.py (hunk 2 above)
Comment (Lunatixz) [reply plugin.video.slingtv/resources/lib/slingtv.py]: thanks, correcting now.

File: plugin.video.slingtv/addon.xml (hunk 1 above)
Comment (Lunatixz) [reply plugin.video.slingtv/addon.xml]: Guess, private repos return 404's

File: plugin.video.slingtv/resources/lib/login.py [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Sling.TV.  If not, see <http://www.gnu.org/licenses/>.
+
+# -*- coding: utf-8 -*-
+import os, sys, time, _strptime, datetime, re, traceback, pytz, calendar, random
+import urlparse, urllib, urllib2, socket, json, requests, base64, inputstreamhelper
+import xbmc, xbmcgui, xbmcplugin, xbmcaddon, xbmcvfs
+from itertools import zip, cycle
```
Comment (Rechi) [inline plugin.video.slingtv/resources/lib/login.py]: There is no zip function in itertools.
```suggestion
from itertools import cycle
```

**Comment by Lunatixz** [comment]
@rechi, git source is public... please let me know if any further changes are needed.

**Comment by Rechi** [comment]
Nothing further needed from my side. Maybe someone else can have a look too.

**Comment by Lunatixz** [comment]
@enen92 can you please take a look; thx

File: plugin.video.slingtv/resources/lib/slingtv.py [hunk 8]
```diff
@@ -0,0 +1,789 @@
...
+            xbmcplugin.setResolvedUrl(int(self.sysARG[1]), True, liz)
+        else:
+            liz = xbmcgui.ListItem(name, path=url)
+            xbmcplugin.setResolvedUrl(int(self.sysARG[1]), True, liz)
+
+        # Hack to fix 6 channel audio causing buffering issues
+        while not xbmc.Player().isPlayingVideo():
+            xbmc.Monitor().waitForAbort(0.25)
+
+        if xbmc.Player().isPlayingVideo() and len(xbmc.Player().getAvailableAudioStreams()) > 1:
```
Comment (enen92) [inline plugin.video.slingtv/resources/lib/slingtv.py]: If this is a bug in Kodi shouldn't it be reported instead of just working around it?

**Pull request merged**